decompiler/frontend/binaryninja/handlers/unary.py

```diff
@@ -0,0 +1,93 @@
+                mediumlevelil.MediumLevelILLoadSsa: partial(self.lift_unary_operation, OperationType.dereference),
+                mediumlevelil.MediumLevelILLoadStruct: self._lift_load_struct,
+                mediumlevelil.MediumLevelILLoadStructSsa: self._lift_load_struct,
+                mediumlevelil.MediumLevelILFtrunc: self._lift_ftrunc,
+            }
+        )
+
+    def lift_unary_operation(self, op_type: OperationType, operation: MediumLevelILOperation, **kwargs) -> UnaryOperation:
+        """Lift the given constant value."""
+        return UnaryOperation(
+            op_type,
+            [self._lifter.lift(x, parent=operation) for x in operation.operands],
+            vartype=self._lifter.lift(operation.expr_type, parent=operation),
+        )
+
+    def lift_cast(self, cast: mediumlevelil.MediumLevelILUnaryBase, **kwargs) -> UnaryOperation:
+        """Lift a cast operation, casting one type to another."""
+        return UnaryOperation(OperationType.cast, [self._lifter.lift(cast.src, parent=cast)], vartype=self._lifter.lift(cast.expr_type))
+
+    def lift_address_of_field(self, operation: mediumlevelil.MediumLevelILAddressOfField, **kwargs) -> Operation:
+        """Lift the address of field operation e.g. &(eax_#1:1)."""
+        if operation.offset == 0:
+            return self.lift_unary_operation(OperationType.address, operation)
+        return BinaryOperation(
+            OperationType.plus,
+            [
+                UnaryOperation(OperationType.address, [operand := self._lifter.lift(operation.src, parent=operation)]),
+                Constant(operation.offset, vartype=operand.type.copy()),
+            ],
+            vartype=self._lifter.lift(operation.expr_type),
+        )
+
+    def _lift_zx_operation(self, instruction: MediumLevelILInstruction, **kwargs) -> UnaryOperation:
+        """Lift zero-extension operation."""
+        inner = self._lifter.lift(instruction.operands[0], parent=instruction)
+        if isinstance(inner.type, Integer) and inner.type.is_signed:
+            unsigned_type = Integer(size=inner.type.size, signed=False)
+            return UnaryOperation(
+                OperationType.cast,
+                [UnaryOperation(OperationType.cast, [inner], unsigned_type)],
+                vartype=self._lifter.lift(instruction.expr_type),
+            )
+        return self.lift_cast(instruction, **kwargs)
+
+    def _lift_load_struct(self, instruction: mediumlevelil.MediumLevelILLoadStruct, **kwargs) -> UnaryOperation:
+        """Lift a MLIL_LOAD_STRUCT_SSA instruction."""
+        return UnaryOperation(
+            OperationType.dereference,
+            [
+                BinaryOperation(
+                    OperationType.plus,
+                    [
+                        UnaryOperation(OperationType.cast, [self._lifter.lift(instruction.src)], vartype=Pointer(Integer.char())),
```

Comment:
I know that `Pointer(Integer.char())` originates from the old lifter, I still wonder why it is `char`... No changes needed.

Reply:
The char pointer is used to indicate it is byte-sized. However, I agree it seems pretty arbitrary